monai/transforms/spatial/array.py

```diff
@@ -3115,29 +3140,33 @@ def __call__(self, array: NdarrayOrTensor):
         patches = list(zip(*patch_iterator))
         patched_image = np.array(patches[0])
         locations = np.array(patches[1])[:, 1:, 0]  # only keep the starting location
+        del patches  # it will free up some memory if padding is used.
```

Comment:
Hi @ericspod , @wyli ,

Do we really need to manually release the var here? I think we usually don't need to care about it unless it accumulates in some for-loop.

Thanks in advance.

Reply:
Hi @Nic-Ma, that's a fair point but here the patches can be very large (but is not being used anymore). On the other hand, when padding patches_image, a similar size image is being created, so we will end up with 4 large arrays (or list of arrays): array, patches, patches_image, and the padded patches_image. Freeing up one may help to use larger input images (although using del does not guarantee free up the memory as it might not be collected by garbage collector at the same time). Do you see any downside of doing that?

Reply:
It's also an antipattern detected here https://deepsource.io/gh/Project-MONAI/MONAI/issue/PTC-W0043/occurrences?listindex=0&page=2

Reply:
Hi @drbeh ,

Thanks for your explanation. As the CI tests also raised it, can we remove it for now? If really has some memory issue, let's add it back.
What do you think?

Thanks.

Reply:
Sure, submitted the PR. Thanks